src/eko/anomalous_dimensions/as1aem1.py

```diff
@@ -355,3 +356,63 @@ def gamma_nsm(N, sx):
         - 16 * zeta3
     )
     return constants.CF * result
+
+
+@nb.njit(cache=True)
+def gamma_singlet(N, nf, sx):
+    nu = constants.uplike_flavors(nf)
+    nd = nf - nu
+    vu = nu / nf
+    vd = nd / nf
+    e2avg = (nu * constants.eu2 + nd * constants.ed2) / nf
+    e2m = constants.eu2 - constants.ed2
+    e2delta = (nd * constants.eu2 + nu * constants.ed2) / nf
+    gamma_S_11 = np.array(
+        [
+            [
+                nf * e2avg * gamma_gg(),
```

Comment:
Please start to factorize right away: find out the most efficient number of independent variables and immediately compute the in a separate function; e.g. I can see
```
abra = vu * e2m
kadabra = vd * e2m
```
is repeated here and below - and it is the atomic ingredient; so only comute them once (you can put, e.g., next to the `uplike_flavors` function) (remember you can return a tuple)

PS: I hope you can find better names then the ones I suggested :see_no_evil:

Reply:
- how about calling `nfe2avg` just `e2tot`?
- and then of course the important part is to split the code from here and put in an extra function that you reuse, e.g. in `gamma_valence`